### PR TITLE
python3Packages.pyemvue: init at 0.18.0

### PR DESCRIPTION
**Pull request by presto8**
## Description of changes

New derivation to add the PyEmVue library for Python. PyEmVue is a Python Library for reading data from the Emporia Vue energy monitoring system.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Tested by using pythonImportsCheck = [...] and verifying build works, and also testing that library functions properly when used with home-assistant service.
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pyemvue/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyemvue/default.nix:20]:
```suggestion
  pyproject = true;
```

we changed that slightly recently but behind the backs it functions the same

File: pkgs/development/python-modules/pyemvue/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyemvue/default.nix:39]:
```suggestion

  # has no tests
  doCheck = false;

```

File: maintainers/maintainer-list.nix
Comment (mweinelt) [inline maintainers/maintainer-list.nix:1]: Adding your maintainer handle should be its own commit: `maintainers: add presto8`.

File: pkgs/development/python-modules/pyemvue/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/pyemvue/default.nix:24]:
```suggestion
    hash = "sha256-s1Uuoiog3OkqOb6PSTqh79vgyLDGFIuZMCyQKH+qG3g=";
```

No functional change now, but allows us to migrate the hash to something else later on.

File: pkgs/development/python-modules/pyemvue/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/pyemvue/default.nix:29]:
```suggestion
```

We always provide `build`, so you can drop it.

File: pkgs/development/python-modules/pyemvue/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/pyemvue/default.nix:42]:
```suggestion
  meta = with lib; {
    changelog = "https://github.com/magico13/PyEmVue/releases/tag/v${version}";
```

**Review by mweinelt — DISMISSED** [review]

File: pkgs/development/python-modules/pyemvue/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/pyemvue/default.nix:37]:
```
Requires-Dist: pycognito>=2021.3.1
Requires-Dist: python-dateutil>=2.8.2
Requires-Dist: requests>=2.26.0
Requires-Dist: typing-extensions>=4.0.1
```

- remove `python-jose`
- rename `dateutil` to `python-dateutil`, also in the function argument

File: pkgs/development/python-modules/pyemvue/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pyemvue/default.nix:44]:
```suggestion
    description = "A Python Library for reading data from the Emporia Vue energy monitoring system";
```
see https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes

File: pkgs/development/python-modules/pyemvue/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pyemvue/default.nix:10]:
```suggestion
, python-dateutil
```

File: pkgs/development/python-modules/pyemvue/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pyemvue/default.nix:47]: You need to add yourself to `maintainers/maintainer-list.nix` in a separate commit.

File: pkgs/development/python-modules/pyemvue/default.nix
Comment (presto8) [reply pkgs/development/python-modules/pyemvue/default.nix]: Thanks! I am not sure why I added python-jose, but how did you produce the report that was included? It looks like it lists the actual imports versus what was specified.

**Comment by presto8** [comment]
Blocked by https://github.com/NixOS/nixpkgs/pull/285686 if someone would be so kind as to review that one :) Thank you!

**Review by dotlambda — APPROVED** [review]

**Comment by mweinelt** [comment]
@ofborg build python311Packages.pyemvue
@ofborg build python312Packages.pyemvue

**Pull request merged**
